src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -52,34 +53,37 @@
 import java.lang.reflect.code.type.JavaType;
 import java.lang.reflect.code.TypeElement;
 import java.lang.reflect.code.type.VarType;
-import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.BitSet;
-import java.util.Deque;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import static java.lang.constant.ConstantDescs.*;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.code.Quotable;
+import java.util.stream.Stream;
+
 /**
  * Transformer of code models to bytecode.
  */
 public final class BytecodeGenerator {
 
-    final MethodHandles.Lookup lookup;
-    final CodeBuilder cob;
-    final Map<Object, Label> labels;
-    final Set<Block> catchingBlocks;
-    final Map<Value, Slot> slots;
+    private static final DirectMethodHandleDesc DMHD_LAMBDA_METAFACTORY = MethodHandleDesc.ofMethod(
+            DirectMethodHandleDesc.Kind.STATIC,
+            LambdaMetafactory.class.describeConstable().orElseThrow(),
+            "metafactory",
+            MethodTypeDesc.of(CD_CallSite, CD_MethodHandles_Lookup, CD_String, CD_MethodType,
+                              CD_MethodType, CD_MethodHandle, CD_MethodType));
 
-    private BytecodeGenerator(MethodHandles.Lookup lookup, Liveness liveness, CodeBuilder cob) {
-        this.lookup = lookup;
-        this.cob = cob;
-        this.labels = new HashMap<>();
-        this.slots = new HashMap<>();
-        this.catchingBlocks = new HashSet<>();
-    }
+    private static final DirectMethodHandleDesc DMHD_LAMBDA_ALT_METAFACTORY = MethodHandleDesc.ofMethod(
+            DirectMethodHandleDesc.Kind.STATIC,
+            LambdaMetafactory.class.describeConstable().orElseThrow(),
+            "altMetafactory",
+            MethodTypeDesc.of(CD_CallSite, CD_MethodHandles_Lookup, CD_String, CD_MethodType,
+                              CD_Object.arrayType()));
```

Comment:
I suppose you could use `ConstantDescs.ofCallsiteBootstrap`, up to you.

Reply:
Good point, it is a bit simpler, thanks.